pkg/scan/scanner.go

```diff
@@ -2,17 +2,23 @@ package scan
 
 import (
 	"fmt"
+	"path/filepath"
 
 	"github.com/hashicorp/terraform-config-inspect/tfconfig"
+	"github.com/keilerkonzept/terraform-module-versions/pkg/modulecall"
 )
 
+// if the local module search goes more than this number, it will stop searching.
+const MaxRecursionDepth int = 10
```

Comment:
let's detect "already-seen" modules instead of globally limiting recursion depth. since a local module (resolved to an absolute path) uniquely identifies a module, we could do a normal graph reachability computation to get all referenced local modules.

Reply:
Hey @sgreben, I've updated the implementation and left some inline comments to help explain the behaviour.

Reply:
Hey @sgreben, just following up on this PR.

Thanks :-)

Reply:
Hey @sgreben, just following up on this PR.